File: client/lib/model/budget_model.dart
class WeddingBudget {
  final int id;
  final int weddingId;
  final int categoryId;
  final double amount;
  final double amountPaid;
  final bool paid;

  WeddingBudget({
    required this.id,
    required this.weddingId,
    required this.categoryId,
    required this.amount,
    this.amountPaid = 0,
    this.paid = false,
  });

  factory WeddingBudget.fromJson(Map<String, dynamic> json) {
    return WeddingBudget(
      id: json['id'],
      weddingId: json['wedding_id'],
      categoryId: json['category_id'],
      amount: (json['amount'] as num).toDouble(),
      amountPaid: (json['amount_paid'] as num).toDouble(),
      paid: json['paid'] ?? false, // Default to false if not provided
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'wedding_id': weddingId,
      'category_id': categoryId,
      'amount': amount,
      'amount_paid': amountPaid,
      'paid': paid,
    };
  }
}

File: server/model/budgetModel.go
package model

import (
	"time"
)

type WeddingBudget struct {
	ID         uint      `json:"id" gorm:"primary_key;autoIncrement"`
	WeddingID  uint      `json:"wedding_id"`
	CategoryID uint      `json:"category_id"`
	Amount     float64   `json:"amount"`
	AmountPaid     float64   `json:"amount_paid"`
	Paid       bool      `json:"paid"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
	DeletedAt  time.Time `json:"deleted_at,omitempty"`
}
